src/models/config.rs

```diff
@@ -157,11 +169,47 @@ impl Repo {
                 target_theme.version = theme.version.clone();
                 target_theme.leftwm_versions = theme.leftwm_versions.clone();
                 target_theme.dependencies = theme.dependencies.clone();
+                target_theme.directory = theme.directory.clone();
             }
             // o/w insert a new leaf at the end
             None => {
                 repo.themes.push(theme.clone());
             }
         }
     }
+
+    // Given a leftwm config path and a theme, checks if the theme is installed
+    // by looking for the theme directory and returns the directory path.
+    fn installed_path(config_path: String, name: String) -> Option<PathBuf> {
```

Comment:
Yea, I do tend to agree that this might be better as `theme.install_path()` in `Theme` rather than `Repo`. What could be helpful here is an `installed_themes() -> Vec<Theme>` (or maybe `Vec<PathBuf>`) to find things that have been installed but not by `leftwm-theme` and are therefore not documented in the Local repo. :)

Reply:
I gave the second approach a try and I think it's a much better approach. Made some changes, instead of returning path, I tried just returning the theme name. So `installed_themes()` returns a list of theme names and in `compare()`, each theme is checked to be present in the installed theme list. If they do, construct a path to their directory and update the `Theme.directory`.
This seems to work well. Also added new tests for this new approach.